File: scripts/fixture.py
#!/usr/bin/env python3
"""Emit a portable, deterministic v1 or v2 fixture (no renderer dependency)."""
import argparse
import json
from pathlib import Path

parser = argparse.ArgumentParser(description=__doc__)
parser.add_argument("--protocol", type=int, choices=[1, 2], default=1)
parser.add_argument("--frame", choices=["first", "second", "all"], default="all")
args = parser.parse_args()
fixtures = Path(__file__).resolve().parent.parent / "fixtures"
name = "home-frame.ndjson" if args.protocol == 1 else "presentation-v2.ndjson"
for line in (fixtures / name).read_text().splitlines():
    message = json.loads(line)
    if message["type"] == "hello":
        message["assetRoot"] = str(fixtures)
    elif args.frame != "all" and message["frame"] != (1 if args.frame == "first" else 2):
        continue
    print(json.dumps(message, separators=(",", ":")), flush=True)

File: scripts/fixture.php
#!/usr/bin/env php
<?php

declare(strict_types=1);
namespace Ichiloto\Renderer\Tools;
require_once __DIR__ . '/lib/Support.php';

runCli(function (): void {
    $args = parseOptions(['protocol' => '1', 'frame' => 'all'], [],
        'Emit deterministic renderer NDJSON. Usage: php scripts/fixture.php [--protocol 1|2] [--frame first|second|all]');
    requireCondition($args['_'] === [] && in_array($args['protocol'], ['1', '2'], true), 'Expected --protocol 1 or 2');
    foreach (getFixtureMessages((int)$args['protocol'], $args['frame']) as $message) {
        echo encodeJson($message), "\n";
        flush();
    }
});
